addons/web_editor/static/src/js/wysiwyg/widgets/link.js

```diff
@@ -502,17 +503,27 @@ const Link = Widget.extend({
      * @param {boolean} force
      */
     _updateLinkContent($link, linkInfos, { force = false } = {}) {
+        const getFirstNotZWSChildElement = (element) => {
+            for(const child of element.children){
+                if (!child.hasAttribute('data-o-link-zws')){
+                    return child;
+                }
+            }
+            return null;
+        };
         if (force || (this.needLabel && (linkInfos.content !== this.data.originalText || linkInfos.url !== this.data.url))) {
             if (linkInfos.content === this.data.originalText) {
                 $link.html(this.data.originalHTML);
             } else if (linkInfos.content && linkInfos.content.length) {
                 let contentWrapperEl = $link[0];
-                // Update the first child element that has the same inner text
+                const text = $link[0].innerText.replaceAll(ZWS_REGEX, "").trim();
```

Comment:
If you use `replaceAll` you don't need a global regex.
So you don't actually need a regex at all:
```suggestion
                const text = $link[0].innerText.replaceAll('\u200B', '').trim();
```